plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -1071,4 +1131,111 @@ public <T> T calling(Callable<T> callable)
             this.statsGetter = requireNonNull(statsGetter, "statsGetter is null");
         }
     }
+
+    private class MetadataCacheAssertions
+    {
+        private final CachingJdbcClient cachingJdbcClient;
+        private final ConnectorSession session;
+        private final SchemaTableName firstName;
+        private final SchemaTableName secondName;
+
+        public MetadataCacheAssertions(CachingJdbcClient cachingJdbcClient, SchemaTableName firstName, SchemaTableName secondName)
+        {
+            this.cachingJdbcClient = requireNonNull(cachingJdbcClient, "cachingJdbcClient is null");
+            this.firstName = requireNonNull(firstName, "firstName is null");
+            this.secondName = requireNonNull(secondName, "secondName is null");
+            session = createSession("asession");
+        }
+
+        public void assertLoadsSchemaAndTableNames()
+        {
+            assertSchemaNamesCache(cachingJdbcClient).loads(1).misses(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getSchemaNames(session)).contains(schema);
+            });
+            assertTableNamesCache(cachingJdbcClient).loads(1).misses(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableNames(session, Optional.empty())).contains(firstName);
+            });
+            assertTableHandleByNameCache(cachingJdbcClient).misses(1).loads(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableHandle(session, firstName)).isNotEmpty();
+            });
+            assertTableHandleByNameCache(cachingJdbcClient).misses(1).loads(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableHandle(session, secondName)).isEmpty();
+            });
+        }
+
+        public void assertCachedSchemaAndTableNames(boolean cacheMissing)
+        {
+            assertSchemaNamesCache(cachingJdbcClient).hits(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getSchemaNames(session)).contains(schema);
+            });
+            assertTableNamesCache(cachingJdbcClient).hits(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableNames(session, Optional.empty()))
+                        .contains(firstName)
+                        .doesNotContain(secondName);
+            });
+            assertTableHandleByNameCache(cachingJdbcClient).hits(1).afterRunning(() -> {
+                assertThat(cachingJdbcClient.getTableHandle(session, firstName)).isNotEmpty();
+            });
+            if (cacheMissing) {
```

Comment:
Since `cacheMissing` works only at a table level - do we need to mix up with the schema related tests. We could configure some default value of schema cache ttl and create a JDBC client. WDYT ?

Reply:
I'm testing a usecase where a newly added table gets reflected in metadata lists faster than the general table cache ttl.